### ISSUE TITLE
Bug in `node_time_indices`, when temporal_block is not included in the model

### ISSUE THREAD

**Issue by mihlema** [Spine-project/SpineOpt.jl#331]
Currently, `node_time_indices` fails, if the node is connected to a `temporal_block`, which is not part of the model. The function call looks like this:
```
function node_time_indices(m::Model; node=anything, temporal_block=anything, t=anything)
    unique(
        (node=n, t=t1) for (n, tb) in node__temporal_block(node=node, temporal_block=temporal_block, _compact=false)
        for t1 in time_slice(m; temporal_block=tb, t=t)
    )
end
```
`time_slice(m; temporal_block=tb, t=t)` throws an error as it doesn't find the `temporal_block` in the `time_slices`. Could `time_slice` simply return an empty array, if the block key doesn't exist in the TimeSliceSet? The possible use case would be to try two different temporal_strucutre on the same model subsequently or exclude/include some parts of the model and compare. By (dis-)connecting the temporal_blocks to the model either structure is applied to the model. Thoughts @manuelma ?

**Comment by DillonJ** [comment]
This is what I was trying to do with the is_active parameters. You should be able to set the is_active parameter of the temporal_block you want to exclude from the run to false. 

But to do this we need to use the tool/feature system. The template already has the "object_activity_control" tool which is linked to all the is_active parameters.

@manuelma Can we use this when calling run_spineopt directly from a Julia script - I don't believe I ever got it to work - but this was a while ago now when all the filtering stuff was still under discussion

**Comment by mihlema** [comment]
I just feel it's a bit untidy at the moment. At the moment we rely on the information that we have a separation between things that are connected to the model, and which aren't. I feel it would be more concise if we also consider the cases where a temporal_block is disconnected from a model.

The `is_active` parameters seem interesting but I also fear that they can come with some problems. If I understand it correctly, they'd need to be interpreted directly within SpineInterface, and are not "readout" into SpineOpt? I don't think I like that.

Would there be a way, to find a generalized way using alternatives somehow?

**Comment by mihlema** [comment]
This has been fixed with 9ee565971345f7b7057496838dfae11d8117bbb7
